### DISCUSSION TITLE
Calibration Tool for Retract

### DISCUSSION THREAD

**Discussion by ClamsyHub** [OrcaSlicer/OrcaSlicer#922] [Ideas]
Hello,
I always used the retract test from http://www.retractioncalibration.com/
See additional hints at: https://www.cnx-software.com/2020/07/08/3d-printer-retraction-calibration-vol-ii-calibration-generator-program-release/

Especially in the comments.

Unfortunately I can't get it to work for my Bambu X1C.

But this retract test is just great. Maybe you can contact the developer and include the tool? Then you could also offer this via the calibration.

Contact person is probably: Karl Johnson
https://www.cnx-software.com/author/test_writer/

**Comment by ClamsyHub** [comment]
Hi,
thanks for the Retraction Test in 1.6.3, but this one is much better. Why this one is not choosen?

**Comment by dewi-ny-je** [comment]
I think this idea should be considered, since it's much better than the tower currently offered in Orca

See how easily the best speed/distance combination can be visually identified

![Capture](https://github.com/SoftFever/OrcaSlicer/assets/2866139/7861ab9d-e50b-4ab6-8d92-6e082ba36c89)

**Comment by inspectionsbybob** [comment]
I like the box, but it takes a while to print, and is very hard to setup correctly, and when it fails, it fails spectacularly.  Some automation is needed in the setup/config, but it is one of the best combo tests...

**Comment by evilDave** [comment]
Just a note: the tower is great, but it needs more instructions for how to pick the "best" part. I think it would be a great addition, particularly if it was added and included some of the feedback ive seen proposed for the tower:
- allow tuning of de-retraction speed
- allow tuning of extra length on restart
- allow tuning of z-hop
(only one at a time obviously)

would it be possible to start the test so that it "brackets" the current value selected in the slicer? (eg if your retraction is 4mm at 60mm/s you get a test the re-tests say 3mm to 6mm and 40mm to 80mm) I think that would make it very effective for tuning in the setting.

And ultimate fine tuning kind of needs to be done on a model that produces more stringing, so leaving the current one there as "step 2" would probably be a good idea too.
